Add files via upload (#279)
Oneflow-Inc/oneflow-documentation: en/docs/extended_topics/user_op.md

* `OpKernelRegistry`：Manage kernel registrations for performing user logic at runtime.

> In programming, you're actually writing your custom Op in C++ and generating the DLL (so file) which you load in Python to use the custom op. You can use the custom op  by loading the corresponding so file in Python. This is the same mechanism that is used in [Extending Op with Python](./python_kernel_op.md). But   the details are encapsulated in the relevant APIs and are transparent to the developer.

Reviewer: ```suggestion
In programming, you're actually writing your custom Op in C++ and generating the DLL (so file) which you load in Python to use the custom op. You can use the custom op by loading the corresponding so file in Python. The mechanism used in [Extending Op with Python](./python_kernel_op.md) is same but the details of that are transparent to the developers.
```

} // namespace oneflow
```

> It does exactly the same thing as [Extending Op with Python](./python_kernel_op.md) that is registers an Op named `myrelu` with `REGISTER_USER_OP` and sets both input output then derives the shape and data type of the output from the input.

Reviewer: ```suggestion
It does exactly the same thing as [Extending Op with Python](./python_kernel_op.md): registers an Op named `myrelu` using `REGISTER_USER_OP` and sets the input, output, and then sets the shape and data type of the output according to the input.
```

In the above code: `flow.config.load_library("final_relu.so") ` is to load the so file.

Building the python wrapper inside `myrelu` is exactly the same as the [ Python interface for wrapping Op](./python_kernel_op.md) in Extending Op with Python. 

So far, we have finished building the `myrelu` which is a relatively simple op. But if we need to build a more complex op, we will need to use some additional advanced features in the registration process.

> We will cover from Op registration, kernel registration, gradient registration and encapsulation of the Python layer.

Reviewer: ```suggestion
We will cover the topics about op registration, kernel registration, gradient registration and Python wrapper.
```

## Detailed Introduction of OpGradRegistry

In [Extending Op with Python](./python_kernel_op.md#op_2), we explained how to provide backwards computation for a custom Op which is registered by the `REGISTER_USER_OP_GRAD`.

> In fact, `REGISTER_USER_OP_GRAD` is actually defining the backward subgraph for derivation. So we don't necessarily need to specifically implement a backward Op to evaluate the gradient which like in [Providing Backward Computation for Custom Op](./python_kernel_op.md#op_2) to implement a backward Op specifically to derive the gradient. But most of the time we can use OneFlow's existing Op to describe the backward subgraph. This section supplements [Using Python Extension Op](./python_kernel_op.md#op_2) by detailing the method of backward registration and representing an inverse subgraph using an existing Op.

Reviewer: ```suggestion
In fact, `REGISTER_USER_OP_GRAD` is actually defining the backward subgraph for derivation. So we don't necessarily need to specifically implement a backward Op to evaluate the gradient which like in [Providing Backward Computation for Custom Op](./python_kernel_op.md#op_2) to implement a backward Op specifically to derive the gradient. Most of the time we can use OneFlow's existing Op to describe the backward subgraph. This section supplements [Using Python Extension Op](./python_kernel_op.md#op_2) by introducing how to construct the backward subgraph using existing op during the gradient registration.
```
